### ISSUE TITLE
Automatically resize window to 1:1 pixel perfect while playing videos

### ISSUE THREAD

**Issue by ChowDPa02k** [jellyfin-archive/jellyfin-desktop-qt#582]
**Is your feature request related to a problem? Please describe.**
The JMP won't adjust window size when playing media. I have to perform complex steps to resize the window to perfect 1920*1080.

I opened an 1080P video on potplayer, then manually resize the JMP window in pixel to totally the same as potplayer, helped by Windows A11y tool "Magnifier" - yes, I have to zoom system in 2x even 3x to make 1 pixel changes.

![image](https://github.com/jellyfin/jellyfin-media-player/assets/16922893/a79ace7b-3027-4dc8-bacc-a6292aafe4d6)

Otherwise, the playback will suffer from upscale / downscale algorithm, blurring.

**Describe the solution you'd like**
An option to enable auto resize window
Just like potplayer and VLC.

**Describe alternatives you've considered**
No

**Additional context**
- If window size smaller than media resolution, resize
- If media resolution larger than display, ignore or maximize
- If window size bigger than media resolution, ignore
- [Optional] A shortcut or menu to resize window to 1/2x 1x 2x, etc.

**Comment by Atmaweaponvi** [comment]
Yeah this feature would be extremely useful with software based upscalers becoming more popular. I'm trying to do the same thing and it appears we're not albe to make the player resize to the video resolution.

**Comment by ChowDPa02k** [comment]
> Yeah this feature would be extremely useful with software based upscalers becoming more popular. I'm trying to do the same thing and it appears we're not albe to make the player resize to the video resolution.

I've achieved that on Windows with 3rd party tools, see https://github.com/jellyfin/jellyfin-media-player/discussions/731 if it helps

**Comment by andrewrabert** [comment]
This Qt desktop client has been deprecated in favor of a rewrite: https://github.com/jellyfin/jellyfin-desktop
